CIScripts/Test/TestRunner.ps1

```diff
@@ -46,29 +47,24 @@ function Invoke-TestScenarios {
         "vRouterAgentService.Tests.ps1"
     )
 
-    $TotalFailedCount = 0
-
     $TestPaths = Get-ChildItem -Recurse -Filter "*.Tests.ps1"
-    Foreach ($TestPath in $TestPaths) {
-        if ($TestPath.Name -in $TestsBlacklist) {
-            Write-Host "Skipping $($TestPath.Name)"
-            continue
-        }
-
-        $PesterRunScript = @{
-            Path=$TestPath.FullName; 
+    $WhitelistedTestPaths = $TestPaths | Where-Object { !($_.Name -in $TestsBlacklist) }
+    $PesterScripts = $WhitelistedTestPaths | ForEach-Object {
+        @{
+            Path=$_.FullName;
             Parameters= @{
                 TestbedAddr=$Sessions[0].ComputerName;
                 ConfigFile=$TestConfigurationFile
             }; 
             Arguments=@()
         }
-
-        $Results = Invoke-Pester -PassThru -Script $PesterRunScript
-        $TotalFailedCount += $Results.FailedCount
     }
-    Write-Host "Num failed tests: $TotalFailedCount"
-    if ($TotalFailedCount -gt 0) {
+    $TestReportOutputPath = "$TestReportOutputDirectory\testReport.xml"
+    $Results = Invoke-Pester -PassThru -Script $PesterScripts `
+        -OutputFormat NUnitXml -OutputFile $TestReportOutputPath
+    Write-Host "Num failed tests: $($Results.FailedCount)"
+    Write-Host "Report written to $TestReportOutputPath"
```

Comment:
Period should be placed at the end of a sentence.
Please note: this is low priority issue.

Reply:
PM: Please add also information about number of passed tests.